### PR TITLE
[WORK IN PROGRESS - DONT MERGE] Zero RPC Handler

### PR DESCRIPTION
**Pull request by RickCarlino**
# What's new?

 * Added implementation to `zero` CS handler
 * Sends new F84 GCode (will update with link to fork of Firmware that has an F84 handler here in a sec)

### CODE REVIEW

File: lib/farmbot/celery_script/commands/zero.ex [hunk 1]
```diff
@@ -8,13 +8,17 @@ defmodule Farmbot.CeleryScript.Command.Zero do
   @behaviour Command
```
Comment (ConnorRigby) [inline lib/farmbot/celery_script/commands/zero.ex]: Going to need: `alias Farmbot.Serial.Handler, as: UartHan` up here somewhere.

File: lib/farmbot/celery_script/commands/zero.ex [hunk 2]
```diff
@@ -8,13 +8,17 @@ defmodule Farmbot.CeleryScript.Command.Zero do
...
-    args: %{axis: axis},
-    body: []
+    Set axis current position to "0":
+      args: %{axis: "x" | "y" | "z" | "all"}
+      body: []
   """
-  @spec run(%{axis: any}, []) :: no_return
-  def run(%{axis: _axis}, []) do
-    Logger.warn "ZERO IS TODO"
+  @spec run(%{axis: String.t}, []) :: no_return
```
Comment (ConnorRigby) [inline lib/farmbot/celery_script/commands/zero.ex]: I have a typespec for axis in `Farmbot.CeleryScript.Command.Home` I can fix this one later tho.

File: lib/farmbot/celery_script/commands/zero.ex [hunk 3]
```diff
@@ -8,13 +8,17 @@ defmodule Farmbot.CeleryScript.Command.Zero do
...
+  @lint {Credo.Check.Refactor.PipeChainStart, false}
+  def run(%{axis: axis}, []) do
```
Comment (ConnorRigby) [inline lib/farmbot/celery_script/commands/zero.ex]: Could we refactor this to not have to not lint this?

``` elixir
def run(%{axis: axis}, []), do: do_write(axis)

@spec do_write(binary) :: no_return
defp do_write("x"), do:"F84 X1 Y0 Z0" |>  UartHan.write
defp do_write("y"), do:"F84 X0 Y1 Z0" |>  UartHan.write
defp do_write("z"), do:"F84 X0 Y0 Z1" |>  UartHan.write
```

or something like that

Comment (RickCarlino) [inline lib/farmbot/celery_script/commands/zero.ex]: :+1: :sparkles:

**Comment by ConnorRigby** [comment]
looks good to me!

**Pull request merged**